src/features/common/Layout/BulkCodeContext.tsx

```diff
@@ -0,0 +1,230 @@
+import {
+  useContext,
+  createContext,
+  useMemo,
+  useState,
+  FC,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+  useCallback,
+} from 'react';
+import {
+  getAuthenticatedRequest,
+  getRequest,
+} from '../../../utils/apiRequests/api';
+import { BulkCodeMethods } from '../../../utils/constants/bulkCodeConstants';
+import { ErrorHandlingContext } from './ErrorHandlingContext';
+import { UserPropsContext } from './UserPropsContext';
+import { TENANT_ID } from '../../../utils/constants/environment';
+import i18next from '../../../../i18n';
+
+const { useTranslation } = i18next;
+
+export interface PlanetCashAccount {
+  guid: string;
+  currency: string;
+  country: string;
+}
+
+export interface Project {
+  guid: string;
+  slug: string;
+  name: string;
+  unitCost: number;
+  currency: string;
+  unit?: string;
+  purpose: string;
+  allowDonations: boolean;
+}
+
+interface BulkGiftGenericData {
+  comment?: string;
+  quantity: number;
+  value: number;
+}
+
+export interface Recipient {
+  units: number;
+  recipientName: string;
+  recipientEmail: string;
+  message: string;
+  notifyRecipient: boolean;
+  // occasion: string;
+}
+
+export interface BulkGiftImportData {
+  comment?: string;
+  occasion: string;
+  recipients: Recipient[];
+}
+
+type BulkGiftData = BulkGiftGenericData | BulkGiftImportData;
+
+export type SetState<T> = Dispatch<SetStateAction<T>>;
+
+interface BulkCodeContextInterface {
+  bulkMethod: BulkCodeMethods.GENERIC | BulkCodeMethods.IMPORT | null;
```

Comment:
This can be written as `BulkCodeMethods | null`. Same goes for line number 69 and 91

Reply:
done